### PR TITLE
Added fields serializer for LinkPreviewOptions

### LINKED ISSUE #1450: Error serializing to JSON: PydanticSerializationError: Unable to serialize unknown type: <class 'aiogram.client.default.Default'>
### Checklist

- [X] I am sure the error is coming from aiogram code
- [X] I have searched in the issue tracker for similar bug reports, including closed ones

### Operating system

Windows 11

### Python version

3.11

### aiogram version

3.4.1

### Expected behavior

Correctly serialized json of Message object

### Current behavior

Seems like aiogram doesn't properly describe pydantic Default class for link_preview_options.

### Steps to reproduce

1. Send message to bot with link preview
2. Try to call model_dump_json method on it

### Code example

```python3
>>> msg.model_dump(exclude_none=True) # default is python mode
 ... link_preview_options=LinkPreviewOptions(is_disabled=<Default('link_preview_is_disabled')>, url='http://t.me/ffffff', ...

>>> msg
 ... link_preview_options=LinkPreviewOptions(is_disabled=<Default('link_preview_is_disabled')>, url='http://t.me/ffffff', prefer_small_media= ...

>>> msg.model_dump_json(exclude_none=True) #trying to serialize to json
pydantic_core._pydantic_core.PydanticSerializationError: Error serializing to JSON: PydanticSerializationError: Unable to serialize unknown type: <class 'aiogram.client.default.Default'>
```

### Logs

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by DanZ-ix**
# Description

Added custom fields json serializer for LinkPreviewOptions, so model_dump_json() works with messages with links

Fixes #1450

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

[x] Send message to bot with link preview
     Try to call model_dump_json method on it
     Message serializes

**Test Configuration**:
* Operating System: Windows 10
* Python version: 3.10

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: aiogram/types/link_preview_options.py [hunk 1]
```diff
@@ -22,6 +24,16 @@ class LinkPreviewOptions(TelegramObject):
...
 
+    @field_serializer(
+        "is_disabled",
+        "prefer_small_media",
+        "prefer_large_media",
+        "show_above_text",
+        when_used="json",
+    )
+    def serialize_fields(self, value):
+        return value.__str__()
```
Comment (JrooTJunior) [inline aiogram/types/link_preview_options.py:35]: You should never call the magic methods directly, just use `str(...)` instead.

File: aiogram/types/link_preview_options.py [hunk 2]
```diff
@@ -22,6 +24,16 @@ class LinkPreviewOptions(TelegramObject):
...
     """*Optional*. :code:`True`, if the link preview must be shown above the message text; otherwise, the link preview will be shown below the message text"""
 
+    @field_serializer(
+        "is_disabled",
+        "prefer_small_media",
+        "prefer_large_media",
+        "show_above_text",
+        when_used="json",
+    )
+    def serialize_fields(self, value):
```
Comment (JrooTJunior) [inline aiogram/types/link_preview_options.py:34]: Typehints required

**Review by JrooTJunior — CHANGES_REQUESTED** [review]
This fix is completely incorrect.

The specified fields are of type `bool`, but for some unknown reason you are converting it to strings like `Default('prefer_small_media')` therefore, the serialization result cannot be properly deserialized.

---

I described the workaround in the issue https://github.com/aiogram/aiogram/issues/1450#issuecomment-2101494052.

But I need to research it more to find a better solution.

**Comment by DanZ-ix** [comment]
Ive wrote my thoughts about this problem in the issue topic, now the Message object can be serialized and deserialized properly

File: aiogram/types/link_preview_options.py (hunk 1 above)
Comment (DanZ-ix) [reply aiogram/types/link_preview_options.py]: fixed

File: aiogram/types/link_preview_options.py (hunk 2 above)
Comment (DanZ-ix) [reply aiogram/types/link_preview_options.py]: fixed

**Comment by JrooTJunior** [comment]
I'm sorry, but you keep making things worse and worse, so stop it.

As I wrote earlier, please wait, I need to analyze the situation and find a system solution, because the problem described is actually not only related to this object.

**Pull request closed without merging**